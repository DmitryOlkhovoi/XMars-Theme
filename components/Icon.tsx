import { omit } from 'underscore';
import React, { FunctionComponent, HTMLAttributes } from 'react';

export interface Props extends HTMLAttributes<HTMLElement> {
    name: string,
    additionalClass?: string
}

const Icon: FunctionComponent<Props> = (props) => {
    const {name, additionalClass} = props;

    return (
       <i className={`icon-${name} ${additionalClass}`} {...omit(props, ['additionalClass', 'name'])}></i>
    );
}

export default Icon;