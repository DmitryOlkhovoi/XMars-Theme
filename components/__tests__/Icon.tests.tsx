import React from 'react';
import Icon from '../Icon';
import renderer from 'react-test-renderer';

describe('Icon component', () => {
  test('Default render', () => {
    const component = renderer.create(
      <Icon name="more" />,
    );
  
    expect(component.toJSON()).toMatchSnapshot();
  });
});