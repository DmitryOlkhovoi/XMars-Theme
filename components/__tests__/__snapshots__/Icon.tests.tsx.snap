// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component Default render 1`] = `
<i
  className="icon-more undefined"
/>
`;
